### DISCUSSION TITLE
Any ideas on how to call @tauri/api from the public folder in react?

### DISCUSSION THREAD

**Discussion by iversonianGremling** [tauri-apps/tauri#7521] [Q&A]
I'm currently working on a synthesizer and wanted to use rust as a backend. I'm using webaudio and a custom AudioWorklet js file to do the job, apparently it HAS to be inside the public folder in react in order to be accessed. I've found success using custom js code but I had trouble when I tried to call/import invoke from the tauri api. Any ideas on how to solve this? Ty in advance.

**Comment by FabianLars** [comment]
I don't think using the npm package is really possible (though i guess that may depend on the bundler you use). Did you try using the global tauri api on the window object already? -> Enable `withGlobalTauri` in tauri.conf.json build config and then access the apis on `window.__TAURI__`

**Reply by iversonianGremling** [reply]
Well, I can't access the object window from there since it's an independent js file in the public folder, here's the code that I'm using rn:


      // public/worklets/audio-processor.js
      const invoke = window.__TAURI__.invoke
      
      class AudioProcessor extends AudioWorkletProcessor {
          process(inputs, outputs, parameters) {
          invoke('generate-sound')
          .then((result) => console.log(result));
      
            const output = outputs[0];
            output.forEach((channel) => {
              for (let i = 0; i < channel.length; i++) {
                channel[i] = 0;
              }
            });
            return true;
          }
        }
      
        registerProcessor("audio-processor", AudioProcessor);
      
        export default AudioProcessor
